src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -90,10 +90,29 @@ def disable_attention_slicing(self):
         # set slice_size = `None` to disable `attention slicing`
         self.enable_attention_slicing(None)
 
-    @torch.no_grad()
+    def gradient_checkpointing_enable(self) -> None:
```

Comment:
This follows the precedent set by `enable_attention_slicing` of providing methods to change this object's configuration.

But it makes for an interface where _most_ of its configuration is done through its constructor arguments, except for these few other things that can only be changed by calling these methods after its creation, which I find to be an awkward mix.

Reply:
Agreed, at least I would make it a constructor argument as well, but I tried to make it similar to what we have in the `transformers` package.

Reply:
gradient checkpointing will be integrated soon #461

Reply:
Awesome! will remove from this PR 👍